### ISSUE TITLE
wrong arguments for Iterator->new() in end() method

### ISSUE THREAD

**Issue by debug-ito** [Leont/list-doublelinked#3]
In end() method, 

``` perl
        return List::DoubleLinked::Iterator->new($self->{tail});
```

This should be 

``` perl
        return List::DoubleLinked::Iterator->new($self, $self->{tail});
```

right?

**Comment by Leont** [comment]
Yeah, that is totally broken.

**Comment by Leont** [comment]
This is fixed in 0.004
